arrow-kt/arrow
# Measure build times

### :pushpin: References
* **Issue:** Fixes #838

### :tophat: What is the goal?

Measure build times as suggested by @raulraja.

### How is it being implemented?

To be able to measure build times we've added a new plugin to this repository enabled for every module. This plugin is named ``build-time-tracker`` and you can find the source code [here](https://github.com/passy/build-time-tracker-plugin). I tried to use [kuronometer](http://kuronometer.io/), but the plugin does not support multiple modules yet. The plugin configuration is the one provided by default in the project documentation but the summary configuration. I've disabled the summary feature because this project has so many modules that the output generated by this plugin could be annoying.

**The build times measured are saved into the ``build`` folder inside a ``times.csv`` file you can review when needed.**

@raulraja, I'm not really sure if this is the result you where expecting when you asked me for a tool to measure the build times, but if this is not the result you expected feel free to discard this PR. 

### How can it be tested?

Build the project executing any Gradle task and you'll see the time execution report and a ``times.csv`` file saved into the root ``build`` folder containing the build execution time information.
